util/ubuntu-setup

```diff
@@ -158,31 +160,21 @@ if [ "$SYSTEM" = "ubuntu-18.04" ]; then
 	apt-mark hold $TOOLCHAIN_PACKAGES
 fi
 
-# Use ubuntu's 'update-alternatives' to select the newer compiler as the default
-# See issue #468 on GitHub
-if [ "$SYSTEM" = "ubuntu-14.04" ]; then
-    update-alternatives --install /usr/bin/g++ g++ $(command -v g++-4.9) 40
-    update-alternatives --set g++ $(command -v g++-4.9)
-fi
-
-if [ "ubuntu-14.04" = "$SYSTEM" -a "$(clang --version | grep -Eo '[0-9]+\.[0-9]+' | head -n1)" \< "3.5" ]; then
-    echo "-- Symlinking clang-3.5 to clang"
-    ln -sf $(which clang-3.5) /usr/bin/clang
-fi
-
 # install python3 requirements
 pip3 install -r $BASE/util/requirements3.txt
 # install python2 requirements
 pip2 install -r $BASE/util/requirements2.txt
 
-# This script is run as sudo, but we don't want submodules to be owned by the
-# root user, so we use `su` to update submodules as the normal user
-echo "-- Updating submodules"
-SETUP_USER="$SUDO_USER"
-if [ -z "$SETUP_USER" ]; then
-    SETUP_USER="$(whoami)"
-fi
+if [ "$NO_SUBMODULES" != "true" ]; then
```

Comment:
At some point, this needs to be done for software as well, the docker hub (I think) downgraded their git version...